examples/advanced/kaplan-meier-he/jobs/kaplan-meier-he/app/config/config_fed_client.conf

```diff
@@ -0,0 +1,116 @@
+{
+  # version of the configuration
+  format_version = 2
+
+  # This is the application script which will be invoked. Client can replace this script with user's own training script.
+  app_script = "kaplan_meier_train.py"
+
+  # Additional arguments needed by the training code. For example, in lightning, these can be --trainer.batch_size=xxx.
+  app_config = "--data_root /tmp/flare/dataset/km_data"
+
+  # Client Computing Executors.
+  executors = [
+    {
+      # tasks the executors are defined to handle
+      tasks = ["train"]
+
+      # This particular executor
+      executor {
+
+        # This is an executor for Client API. The underline data exchange is using Pipe.
+        path = "nvflare.app_opt.pt.client_api_launcher_executor.ClientAPILauncherExecutor"
+
+        args {
+          # launcher_id is used to locate the Launcher object in "components"
+          launcher_id = "launcher"
+
+          # pipe_id is used to locate the Pipe object in "components"
+          pipe_id = "pipe"
+
+          # Timeout in seconds for waiting for a heartbeat from the training script. Defaults to 30 seconds.
+          # Please refer to the class docstring for all available arguments
+          heartbeat_timeout = 60
+
+          # format of the exchange parameters
+          params_exchange_format =  "raw"
+
+          # if the transfer_type is FULL, then it will be sent directly
+          # if the transfer_type is DIFF, then we will calculate the
+          # difference VS received parameters and send the difference
+          params_transfer_type = "FULL"
+
+          # if train_with_evaluation is true, the executor will expect
+          # the custom code need to send back both the trained parameters and the evaluation metric
+          # otherwise only trained parameters are expected
+          train_with_evaluation = false
+        }
+      }
+    }
+  ],
+
+  # this defined an array of task data filters. If provided, it will control the data from server controller to client executor
+  task_data_filters =  []
```

Comment:
Can we do the homomorphic encryption and decryption using filters?

Reply:
moved the context generation and distribution part out of workflow to be a pre-step (and load them in wf and local script), filters will need some modifications to load the specific context file from a given path, we can make the update in a separate PR

Reply:
Why not in this PR? Using filters for HE encryption/decryption would make the code more reusable. Also, this example could be a nice to add support for the BFV scheme to the existing HE filters (which also use TenSEAL). Note, the existing filters already allow reading the ts context from file. See [here](https://github.com/NVIDIA/NVFlare/blob/030b3d253b25e2d6a12f28df22b3f77661a76678/nvflare/app_opt/he/model_decryptor.py#L36).

Reply:
The filter way may not directly work for this particular application, because in different rounds we target different goal: the round to collect max bin number and calculate global max is not supported by current HE.
So for this workflow HE will need to be turn off and on. We need to figure out a way to specify the particular round to apply filter, or break the workflow to several tasks.